homeassistant/components/ebusd/sensor.py

```diff
@@ -19,8 +24,15 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Set up the Ebus sensor."""
+    if not discovery_info:
+        return
```

Comment:
The rest of the code assumes `discovery_info` is available.
Is this the best to resolve this?
Other options include `assert` (with or without TYPE_CHECKING).
Do we maybe need to log something?

Reply:
This is good. The only case when the platform could be setup without discovery info is by a misconfigured platform section in configuration.yaml. We use a check like this as standard when using our old discovery helper to guard against that case.